src/main/java/com/limechain/runtime/hostapi/PartialHostApi.java

```diff
@@ -0,0 +1,34 @@
+package com.limechain.runtime.hostapi;
+
+import org.wasmer.ImportObject;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * An interface defining an implementation of a (not necessarily strict) subset of all {@link Endpoint}s.
+ * @apiNote Mainly intended for partitioning a total implementation into separate classes for readability
+ *          and separation of concerns.
+ */
+public interface PartialHostApi {
```

Comment:
Considering the `HostApi` and the `DefaultHostApi` this name is confusing. Maybe we could switch it to `PartialHostFunctions`

Reply:
The analog to `HostApi` is intentional here. `HostApi` is an abstraction for a total implementation of the API, whilst `PartialHostApi` is an abstraction for a potentially partial implementation. The only difference between the two being the strictness of expectation for the provided `FuncImport`s.

You're right, however, that naming is not consistent. Initially, all `...HostFunctions` classes were renamed to `...HostApi`, but in order to keep diffs readable and preserve git history, I finally kept to the old names.

Now it's a matter of choice - which naming we'd want to choose, and then unify names to that convention. Personally, I like `StorageHostApi` more than `StorageHostFunctions`... Or to be more precise, `DefaultStorageApi` would be the most consistent (having to do with the fact, that we also possibly allow for other implementation; although we probably won't need this abstraction much outside of testing).

Reply:
> Now it's a matter of choice - which naming we'd want to choose, and then unify names to that convention.

This means we would switch all `...HostFunctions` to `...HostApi` if I understand correctly. If that is the case I am okay with that. Considering there are javadocs incode what you explained about `HostApi` vs `PartialHostApi` seems logical.

Reply:
Delegated to the follow-up PR.